kernel/Utilities/SparsityEstimator.cpp

```diff
@@ -15,310 +18,416 @@ SparsityEstimator::SparsityEstimator(const GlobalIndexing& rowindexing,
-                element->isOwnedByCurrentProcessor(),
-                "Face coupling with conforming basis functions on distributed "
-                "meshes is not supported");
-            for (const Base::Face* otherFace : element->getFacesList()) {
-                if (otherFace == face)
-                    continue;  // Both left & right element of the original face
-                               // are already visited
-                if (!otherFace->isInternal()) continue;
-                const Base::Element* otherElement =
-                    otherFace->getPtrOtherElement(element);
-                // The basis function from the face may couple through any other
-                // face of the elements adjacent to the face
-                addElementDoFs(otherElement, workspace);
+                for (Detail::DoFInterval rowInterval : rowIntervals) {
+                    for (Detail::DoFInterval colInterval : ownedColIntervals) {
+                        owned[rowInterval].emplace(colInterval);
+                    }
+                    for (Detail::DoFInterval colInterval :
+                         nonOwnedColIntervals) {
+                        nonOwned[rowInterval].emplace(colInterval);
+                    }
+                }
             }
         }
-        // Store the information
-        writeDoFCount(face, workspace, nonZeroPerRowOwned,
-                      nonZeroPerRowNonOwned);
     }
-    logger(VERBOSE, "Sparsity pattern for Face DoFs computed");
-    // Edges
-    for (const Base::Edge* edge : rowIndexing_.getMesh()->getEdgesList()) {
-        if (!edge->isOwnedByCurrentProcessor() ||
-            !hasLocalDoFs(edge, rowIndexing_.getIncludedUnknowns())) {
-            continue;
-        }
-        workspace.clear();
-        for (const Base::Element* element : edge->getElements()) {
-            // DoFs from supporting element
-            addElementDoFs(element, workspace);
-            // Face matrix coupling with the element on the other side of the
-            // face
-            if (!includeFaceCoupling) continue;
-            // See remark for face based DoFs
-            logger.assert_always(
-                element->isOwnedByCurrentProcessor(),
-                "Face coupling with conforming basis functions on distributed "
-                "meshes is not supported");
-            for (const Base::Face* face : element->getFacesList()) {
-                if (!face->isInternal()) continue;
-                const Base::Element* otherElement =
-                    face->getPtrOtherElement(element);
-                addElementDoFs(otherElement, workspace);
-            }
+
+    // If there is no face coupling anywhere (e.g. dg mass matrix) then we can
+    // skip looping over the adjacent elements.
+    bool anyFaceCoupling = false;
+    for (std::size_t i = 0; i < faceCoupling.getSize(); ++i) {
+        if (faceCoupling[i]) {
+            anyFaceCoupling = true;
+            break;
```

Comment:
Can we iterate over facecoupling itself instead of using an index?

Reply:
Maybe, it feels a bit strange to have a begin and end on a 2D table. I'll have a thought about it.